### PR TITLE
Check the update_rate set to the controllers to be a valid one

### PR DESCRIPTION
**Pull request by saikishor**
Right now, if we have set an invalid value to the update_rate (rate ggreater than CM's rate) the internal variable still holds the invalid one. This PR aims to add some checks to detect these situations and set it to the CM's rate.

### CODE REVIEW

File: controller_manager/test/test_load_controller.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ TEST_F(TestLoadController, can_set_and_get_non_default_update_rate)
   EXPECT_EQ(
     lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if->get_lifecycle_state().id());
 
-  EXPECT_EQ(1337u, controller_if->get_update_rate());
```
[the remark this answers is not in the document]
Comment (saikishor) [reply controller_manager/test/test_load_controller.cpp:75]: > Well this assertions is ensuring that setting the node update_rate to something gets propagated :D 
> 
> > controller_if->get_node()->set_parameter({"update_rate", 1337});
> 
> Could you please change that number to something that would pass the current checks and not reset to the CM update_rate?

Sure!

[the remark this answers is not in the document]
Comment (saikishor) [reply controller_manager/test/test_load_controller.cpp:75]: @bmagyar Done

[the remark this answers is not in the document]
Comment (bmagyar) [reply controller_manager/test/test_load_controller.cpp:75]: thank you!

[the remark this answers is not in the document]
Comment (bmagyar) [reply controller_manager/test/test_load_controller.cpp:75]: Ah wait, no! :D 

>   controller_if->get_node()->set_parameter({"update_rate", 1337});

so we should set it to some other number, like

>  controller_if->get_node()->set_parameter({"update_rate", 25});
and check that setting it to that works. 
> EXPECT_EQ(25u, controller_if->get_update_rate());

as the test title is `can_set_and_get_non_default_update_rate`

[the remark this answers is not in the document]
Comment (saikishor) [reply controller_manager/test/test_load_controller.cpp:75]: Sure! Let me handle it

[the remark this answers is not in the document]
Comment (saikishor) [reply controller_manager/test/test_load_controller.cpp:75]: @bmagyar Done. The tests are passing

**Review by bmagyar — APPROVED** [review]

**Pull request merged**